app/src/main/java/com/fmsys/snapdrop/MainActivity.java

```diff
@@ -484,48 +481,39 @@ public void onReceivedError(final WebView view, final WebResourceRequest request
     }
 
     private void copyTempToDownloads(final JavaScriptInterface.FileHeader fileHeader) {
-
         executor.execute(() -> {
-
             final FileDescription fileDescription = new FileDescription(fileHeader.getName(), "", fileHeader.getMime());
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                final MediaFile newFile = MediaStoreCompat.createDownload(this, fileDescription, true);
-                if (newFile != null) {
-                    DocumentFileUtils.moveFileTo(DocumentFile.fromFile(fileHeader.getTempFile()), this, newFile, fileCallback(fileHeader));
-                }
-            } else {
-                DocumentFileUtils.moveFileTo(DocumentFile.fromFile(fileHeader.getTempFile()), this, Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS), fileDescription, fileCallback(fileHeader));
-            }
-
+            final DocumentFile source = DocumentFile.fromFile(fileHeader.getTempFile());
+            DocumentFileUtils.moveFileToDownloadMedia(source, getApplicationContext(), fileDescription, fileCallback(fileHeader));
         });
-
     }
 
     private FileCallback fileCallback(final JavaScriptInterface.FileHeader fileHeader) {
         return new FileCallback() {
             @Override
             public void onFailed(@NotNull final FileCallback.ErrorCode errorCode) {
                 Log.d("SimpleStorage", errorCode.toString());
-                //UI Thread
-                handler.post(() -> Toast.makeText(MainActivity.this, errorCode.toString(), Toast.LENGTH_SHORT).show());
+                Toast.makeText(MainActivity.this, errorCode.toString(), Toast.LENGTH_SHORT).show();
             }
 
             @Override
             public void onCompleted(@NotNull final Object file) {
-                Uri uri = null;
+                final Uri uri;
 
                 if (file instanceof MediaFile) {
                     final MediaFile mediaFile = (MediaFile) file;
                     uri = mediaFile.getUri();
                 } else if (file instanceof DocumentFile) {
                     final DocumentFile documentFile = (DocumentFile) file;
-                    uri = DocumentFileUtils.isRawFile(documentFile) ? FileProvider.getUriForFile(MainActivity.this, getApplicationContext().getPackageName() + ".provider", DocumentFileUtils.toRawFile(documentFile)) : documentFile.getUri();
+                    final Context context = getApplicationContext();
```

Comment:
Thanks for your contribution! Very welcomed 👍

Is there a reason why you have chosen `getApplicationContext` instead of `MainActivity.this`?

Reply:
There's no reason at line 508. But for line 487, it's better to use application context because it contains background processing.